### PR TITLE
Add more tests for message-passing.

### PR DESCRIPTION
**Pull request by djpetti**

### CODE REVIEW

File: common/message_passing/tests/test_integration.cpp [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+ * and produce a response from the server.
+ */
+TEST(MessagePassingIntegration, RequestResponse) {
+  // Arrange.
+  auto config = MakeConfig();
+
+  // Act.
+  // Send a request to the server.
+  const auto kTestRequest = MakeTestMessage();
+  ASSERT_TRUE(Retry([&]() { return config.client->Send(kTestRequest) > 0; }));
```
Comment (Nickster258) [inline common/message_passing/tests/test_integration.cpp:115]: Honestly, this looks like magic. Where should I read up on this?

**Review by Nickster258 — APPROVED** [review]
LGTM

File: common/message_passing/tests/test_integration.cpp (hunk 1 above)
Comment (djpetti) [reply common/message_passing/tests/test_integration.cpp:115]: You mean [lambda expressions](https://en.cppreference.com/w/cpp/language/lambda)?

Comment (jake-chandler) [reply common/message_passing/tests/test_integration.cpp:115]: Yeah, I'm not entirely sure what's going on here either

**Review by jake-chandler — APPROVED** [review]

File: common/message_passing/tests/test_integration.cpp (hunk 1 above)
Comment (Nickster258) [reply common/message_passing/tests/test_integration.cpp:115]: > You mean [lambda expressions](https://en.cppreference.com/w/cpp/language/lambda)?

Yeah I had assumed it was something like that, just didn't expect it to be called a lambda.

**Pull request merged**